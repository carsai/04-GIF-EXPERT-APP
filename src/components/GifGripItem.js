import React from 'react';
import PropTypes from 'prop-types';

export const GifGripItem = ({title, url}) => {

    return (
        <div className='card animate__animated animate__bounce'>
            <img src={url} alt={title}></img>
            <p>{title}</p>
        </div>
    )
}

GifGripItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
