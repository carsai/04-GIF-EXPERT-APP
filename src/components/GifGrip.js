import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGripItem } from './GifGripItem';

export const GifGrip = ({category}) => {

    const {data, loading} = useFetchGifs(category);     

    return (
        <>
            <h3 className='animate__animated animate__bounce'>{category}</h3> 
            {loading && <p className='animate__animated animate__flash'>cargando</p>}
            <div className='card-grid'>
            {
                    data.map(img => <GifGripItem 
                            key={img.id} 
                            {...img}
                        />)
            }
            </div>
        </>
    )
}
