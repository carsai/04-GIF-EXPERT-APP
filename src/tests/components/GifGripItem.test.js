import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';

import { GifGripItem } from '../../components/GifGripItem';

describe('Pruebas con el componente <GifGripItem />', () => {
    
    const title = 'Titulazooo';
    const url    = 'https://media0.giphy.com/media/xjb8iUCuPBKGqbuw4D/giphy.gif?cid=59b9c425y7lgnw8oq5bxgq8i0uo5gda2cow1k871q1h4al39&rid=giphy.gif';
    
    const wrapper = shallow(<GifGripItem title={title} url={url} />);

    test('Debe mostrar el componente GifGripItem correctamente ', () => {

        expect(wrapper).toMatchSnapshot();
        
    });

    test('Debe tener un <p> con el "title"', () => {

        const pText = wrapper.find('p').text().trim();
        
        expect(pText).toBe(title);
        
    });

    test('Debe tener in <img> con "src" y "alt" valido', () => {

        const {src, alt} = wrapper.find('img').props();

        expect(src).toBe(src);
        expect(alt).toBe(title);
        
    });

    test('<div> debe tener la clase "animate__bounce"', () => {

        const div = wrapper.find('div');

        expect(div.hasClass('animate__bounce')).toBe(true);
        
    })
        
});
