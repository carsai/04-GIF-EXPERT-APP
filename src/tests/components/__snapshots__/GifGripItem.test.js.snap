// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el componente <GifGripItem /> Debe mostrar el componente GifGripItem correctamente  1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Titulazooo"
    src="https://media0.giphy.com/media/xjb8iUCuPBKGqbuw4D/giphy.gif?cid=59b9c425y7lgnw8oq5bxgq8i0uo5gda2cow1k871q1h4al39&rid=giphy.gif"
  />
  <p>
    Titulazooo
  </p>
</div>
`;
